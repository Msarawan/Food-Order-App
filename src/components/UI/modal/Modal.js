import React,{ Fragment, useEffect } from 'react';
import ReactDOM from 'react-dom';
import classes from './Modal.module.css';

const Backdrop = (props) => {
  return <div className={classes.backdrop} onClick={props.onClose} />;
};

const ModalOverlay = (props) => {
  return (
    <div className={classes.modal}>
      <div className={classes.content}>{props.children}</div>
    </div>
  );
};


const portalElement = document.getElementById('overlays');
const modalElement = document.createElement('div');

  // appends the modal to portal once modal's children are mounted and 
  // removes it once we don't need it in the DOM anymore:
 
const Modal = (props) => {
 /*  useEffect(() => {
    portalElement.appendChild(modalElement);
    return () => {
      portalElement.removeChild(modalElement);
    };
  }, [modalElement]); */
  return (
    <Fragment>
      {<div>ReactDOM.createPortal(<Backdrop onClose={props.onClose} />, portalElement)</div>}
      
     {<div>ReactDOM.createPortal(
        <ModalOverlay>{props.children}</ModalOverlay>,portalElement )</div>}
       
    </Fragment>
  );
};

export default Modal;