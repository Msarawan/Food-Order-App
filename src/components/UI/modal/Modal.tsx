import * as React from 'react';
import { Fragment} from 'react';
import classes from './Modal.module.css';

interface Props {
 onClose:any,
 children:any
}

const Backdrop = (props:Props) => {
  return <div className={classes.backdrop} onClick={props.onClose} />;
};

const ModalOverlay = (props:Props) => {
  return (
    <div className={classes.modal}>
      <div className={classes.content}>{props.children}</div>
    </div>
  );
};


const portalElement = document.getElementById('overlays');
const modalElement = document.createElement('div');

  // appends the modal to portal once modal's children are mounted and 
  
const Modal = (props: { children: any; }) => {
  return (
    <Fragment>
      {<div>ReactDOM.createPortal(<Backdrop onClose={undefined} children={undefined}/>, portalElement)</div>}
      
     {<div>ReactDOM.createPortal(
        <ModalOverlay>{props.children}</ModalOverlay>,portalElement )</div>}
       
    </Fragment>
  );
};

export default Modal;