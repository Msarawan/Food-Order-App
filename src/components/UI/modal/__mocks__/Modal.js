import React from 'react';
const MockedModal = () => {
  return <></>;
}
export default MockedModal;