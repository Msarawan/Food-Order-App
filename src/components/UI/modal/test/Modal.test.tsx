import React from "react";
import Modal from "../Modal";
import { mount} from "enzyme";
import ReactDOM from 'react-dom';
//import { Backdrop,  ModalOverlay,} from '../Modal';
 

 describe('Modal component', () => {
  let wrapper:any;

    const modalRoot = global.document.createElement('div');
     modalRoot.setAttribute('id', 'overlays');
     const body = global.document.querySelector('body');
     body?.appendChild(modalRoot);

    test('should render the modal wrapper', () => {
      wrapper = mount(
        <Modal onClose={function (): void {
          throw new Error("Function not implemented.");
        } } children={undefined}/>
     );   
       expect(modalRoot.hasChildNodes()).toBeFalsy();
       expect(wrapper.find("Backdrop").exists()).toBeTruthy();
       expect(wrapper.find("ModalOverlay").exists()).toBeTruthy();

       expect(wrapper).toMatchSnapshot();
      })
    })
