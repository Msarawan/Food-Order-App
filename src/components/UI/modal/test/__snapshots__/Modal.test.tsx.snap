// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Modal component should render the modal wrapper 1`] = `
<Modal
  onClose={[Function]}
>
  <div>
    ReactDOM.createPortal(
    <Backdrop
      onClose={[Function]}
    >
      <div
        className="backdrop"
        onClick={[Function]}
      />
    </Backdrop>
    , portalElement)
  </div>
  <div>
    ReactDOM.createPortal(
    <ModalOverlay>
      <div
        className="modal"
      >
        <div
          className="content"
        />
      </div>
    </ModalOverlay>
    ,portalElement )
  </div>
</Modal>
`;
